sdk/core/azure-core/src/main/java/com/azure/core/http/policy/UserAgentPolicy.java

```diff
@@ -35,6 +36,33 @@ public class UserAgentPolicy implements HttpPipelinePolicy {
     public static final String APPEND_USER_AGENT_CONTEXT_KEY = "Append-User-Agent";
 
     private final String userAgent;
+    private final HttpPipelineSyncPolicy inner = new HttpPipelineSyncPolicy() {
```

Comment:
INNER an be static.

Reply:
nvm this.

Reply:
INNER does not need to be static as it might need to access instance variables for when before/after processing of the request.